# Q-Learning Grid Search exploitation & researchs

## Imports

In [24]:
%pip install -r requirements.txt

Note: you may need to restart the kernel to use updated packages.


In [25]:
import os
import pandas as pd
import numpy as np

## Data Importation

In [26]:
PATH = os.getcwd()
DATA_PATH = os.path.join(PATH, "results_grid-search_v2.csv")

In [27]:
df = pd.read_csv(DATA_PATH, header=0, on_bad_lines='skip', sep=';')

In [28]:
df = df.applymap(lambda x: float("{:.2f}".format(x)) if type(x) == float else x)

In [29]:
df

Epsilon  Alpha  Gamma  Epochs_learn  Penalties_learn  Epochs_eval   
0        0.15   0.05   0.35       1953066            89536         1301  \
1        0.15   0.05   0.36       1948198            88953         1310   
2        0.15   0.05   0.37       1939543            89269         1278   
3        0.15   0.05   0.38       1931277            88440         1289   
4        0.15   0.05   0.39       1929364            88603         1328   
...       ...    ...    ...           ...              ...          ...   
1326     0.25   0.15   0.41       1922714           144413         1275   
1327     0.25   0.15   0.42       1918438           144459         1264   
1328     0.25   0.15   0.43       1916916           143594         1280   
1329     0.25   0.15   0.44       1913861           143771         1325   
1330     0.25   0.15   0.45       1913162           143338         1297   

      Penalties_eval  Worked  
0                  0    True  
1                  0    True  
2                  0    True  
3                  0    True  
4                  0    True  
...              ...     ...  
1326               0    True  
1327               0    True  
1328               0    True  
1329               0    True  
1330               0    True  

[1331 rows x 8 columns]

In [30]:
df.groupby(["Epsilon"]).size()

Epsilon
0.15    121
0.16    121
0.17    121
0.18    121
0.19    121
0.20    121
0.21    121
0.22    121
0.23    121
0.24    121
0.25    121
dtype: int64

## Data Exploration

In [31]:
df_worked = df[df["Worked"] == True]

df_worked.shape

(1331, 8)

In [32]:
df_worked["Penalties_eval"].value_counts()

Penalties_eval
0    1331
Name: count, dtype: int64

In [33]:
df_worked_EE_min = df_worked[df_worked["Epochs_eval"] == df_worked["Epochs_eval"].min()]

df_worked_EE_min

Epsilon  Alpha  Gamma  Epochs_learn  Penalties_learn  Epochs_eval   
1011     0.23   0.08   0.45       1956714           135830         1219  \

      Penalties_eval  Worked  
1011               0    True

In [34]:
df_worked_EE_min[df_worked_EE_min["Epochs_learn"] == df_worked_EE_min["Epochs_learn"].min()]

Epsilon  Alpha  Gamma  Epochs_learn  Penalties_learn  Epochs_eval   
1011     0.23   0.08   0.45       1956714           135830         1219  \

      Penalties_eval  Worked  
1011               0    True

## Conclusion

With this little exploration, we can see the best hyperparameters are:

- **Epsilon**: 0.23
- **Alpha**: 0.08
- **Gamma**: 0.45